### DISCUSSION TITLE
renovate@37.192.2 gives me segmentation fault on node 18.19.1

### DISCUSSION THREAD

**Discussion by kornicameister** [renovatebot/renovate#27360] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Cloud (bitbucket.org)

### Was this something which used to work for you, and then stopped?

I have not seen this working

### Describe the problem

I have installed `renovate` on top of `node@18.19.1` and trying to run it gives me only:

```
/shims/renovate: line 43:    38 Segmentation fault      $(nodenv which renovate) "$@"
```

The shims part you see in command is my binary on top of actual renovate that authorizes with AWS CodeArtifact and augments the config.js file with appropriate token and artifact details. That part seems to be working just fine.

```
#!/bin/bash

# make same fail-fast exits
set -o errexit
set -o nounset

# get token from AWS
if [ -n "${BITBUCKET_STEP_OIDC_TOKEN:-""}" ]; then
	echo "[npm] Logging into CodeArtifact via OIDC"

	unset AWS_ACCESS_KEY_ID
	unset AWS_SECRET_ACCESS_KEY
	unset AWS_SESSION_TOKEN
	unset AWS_DEFAULT_REGION
	unset AWS_ASSUME_ROLE_ARN

	tokenFile="$(pwd)/web-identity-token"
	export AWS_ROLE_ARN="${AWS_ROLE_ARN:-"arn:aws:iam::00000000:role/artifact-role"}"
	export AWS_REGION="${AWS_REGION:-"eu-central-1"}"
	export AWS_WEB_IDENTITY_TOKEN_FILE="${tokenFile}"

	echo "${BITBUCKET_STEP_OIDC_TOKEN}" >"${tokenFile}"
	chmod 400 "${tokenFile}"
fi

registryDomain=${AY_REGISTRY_DOMAIN:-"xxxx"}
registryOwner=${AY_REGISTRY_OWNER:-"00000000"}
registryRegion=${AY_REGISTRY_REGION:-"eu-central-1"}
codeartifactToken="$(
	aws codeartifact get-authorization-token \
		--domain "${registryDomain}" \
		--domain-owner "${registryOwner}" \
		--region "${registryRegion}" \
		--query authorizationToken \
		--output text
)"

sed -i "s/<AY_REGISTRY_DOMAIN>/${registryDomain}/g" /renovate/config.js
sed -i "s/<AY_REGISTRY_OWNER>/${registryOwner}/g" /renovate/config.js
sed -i "s/<AY_REGISTRY_REGION>/${registryRegion}/g" /renovate/config.js
sed -i "s/<CODEARTIFACT_AUTH_TOKEN>/${codeartifactToken}/g" /renovate/config.js

$(nodenv which renovate) "$@"
```

### Relevant debug logs

<details><summary>Logs</summary>

```
/shims/renovate: line 43:    38 Segmentation fault      $(nodenv which renovate) "$@"
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment]
Please describe in more detail how you're running Renovate

**Comment by kornicameister** [comment]
@rarkins I have my own docker image that is packed with stuff used by projects in my organization.
One of those things is `renovate` now. Essentially the renovate is being:

- installed with `npm` as in `npm install -g renovate@37.192.2`
- wrapped in `shim` that code is linked above

Image is then used in Bitbucket Pipeline (BB Cloud) and runs on top of this config.js:

```javascript
module.exports = {
  platform: "bitbucket",
  token: process.env.RENOVATE_TOKEN,
  baseDir: `${process.env.BITBUCKET_CLONE_DIR}/renovate`,
  repositories: [`${process.env.BITBUCKET_REPO_FULL_NAME}`],
  repositoryCache: "enabled",
  logLevel: "debug",
  dependencyDashboard: true,
  packageRules: [
    {
      matchDatasources: ["npm"],
      registryUrls: [
        "https://<AY_REGISTRY_DOMAIN>-<AY_REGISTRY_OWNER>.d.codeartifact.<AY_REGISTRY_REGION>.amazonaws.com/",
      ],
    },
  ],
  hostRules: [
    {
      hostType: "npm",
      matchHost:
        "https://<AY_REGISTRY_DOMAIN>-<AY_REGISTRY_OWNER>.d.codeartifact.<AY_REGISTRY_REGION>.amazonaws.com/",
      token: "<CODEARTIFACT_AUTH_TOKEN>",
    },
  ],
};
```

Usage example:

```yml
step:
  name: renovate
  image: my-cicd-image
  caches:
    - renovate
  script:
    - renovate
```

**Comment by kornicameister** [comment]
That's weird...it is able to run once, then again it cannot with segmentation fault, and then again it can.
If it fails, logs just something like `PID 229 received SIGSEGV for address: 0x7f9f5c1f2000`

Final configuration that sometimes make everything start is:

```log
DEBUG: Using RE2 regex engine
DEBUG: Parsing configs
DEBUG: Checking for config file in /renovate/config.js
DEBUG: File config
       "config": {
         "platform": "bitbucket",
         "baseDir": "/renovate",
         "token": "***********",
         "repositories": ["my_repo"],
         "repositoryCache": "enabled",
         "packageRules": [
           {
             "matchDatasources": ["npm"],
             "registryUrls": [
               "https://appyourself-00000000000.d.codeartifact.eu-central-1.amazonaws.com/"
             ]
           }
         ],
         "hostRules": [
           {
             "hostType": "npm",
             "matchHost": "https://appyourself-00000000000.d.codeartifact.eu-central-1.amazonaws.com",
             "token": "***********"
           }
         ]
       }
DEBUG: CLI config
       "config": {}
DEBUG: Env config
       "config": {"hostRules": []}
DEBUG: Combined config
       "config": {
         "platform": "bitbucket",
         "baseDir": "/renovate",
         "token": "***********",
         "repositories": ["my_repo"],
         "repositoryCache": "enabled",
         "packageRules": [
           {
             "matchDatasources": ["npm"],
             "registryUrls": [
               "https://appyourself-00000000000.d.codeartifact.eu-central-1.amazonaws.com/"
             ]
           }
         ],
         "hostRules": [
           {
             "hostType": "npm",
             "matchHost": "https://appyourself-00000000000.d.codeartifact.eu-central-1.amazonaws.com",
             "token": "***********"
           }
         ]
       }
DEBUG: Enabling forkProcessing while in non-autodiscover mode
DEBUG: Found valid git version: 2.39.2
DEBUG: Setting global hostRules
DEBUG: Adding token authentication for https://appyourself-00000000000.d.codeartifact.eu-central-1.amazonaws.com (hostType=npm) to hostRules
DEBUG: hostRules: authentication already set for api.bitbucket.org
```

**Comment by kornicameister** [comment]
Interestingly, if I try to run my code based on official docker image `renovate/renovate:37.192.3-full` I also get segmentation fault but this time with slightly different error: `/usr/local/bin/node: line 6:    16 Segmentation fault      /opt/containerbase/tools/node/18.19.1/bin/node --use-openssl-ca "$@"`

**Reply by rarkins** [reply]
Can you reproduce the problem publicly?

**Reply by kornicameister** [reply]
What do you mean publicly?

**Reply by kornicameister** [reply]
The only way, so far, when I was able to run `renovate` was locally from my own laptop.
It continuously works there correctly and I was able to get onboarding PR up.
But no matter which docker I try (mine or the official one) there's a segmentation fault.

**Reply by kornicameister** [reply]
Ok, I am now 100% puzzled.
I cannot run `renovate` in docker container if I build image locally
But if image is built in CICD and later used in another pipeline, `renovate` works.

**Comment by kornicameister** [comment]
I got renovate working correctly.
I couldn't find an exact problem but the moment I stopped trying to run via docker it started to work.

I don't know what though might have been so different on my laptop that made even official image fail.

**Reply by viceice** [reply, marked as answer]
this seems to be an issue with your environment and not a renovate one.
